ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/ReplLoadTask.java

```diff
@@ -166,14 +185,24 @@ a database ( directory )
           }
           loadTaskTracker.update(functionsTracker);
           functionsTracker.debugLog("functions");
+
+          // Add ReplStateLogTask as function load tasks are always fully added in current cycle
+          loadTaskTracker.update(createFunctionReplLogTask(functionsTracker.tasks(), maxTasks,
+                  iterator.replLogger(), loadFunction.functionName()));
```

Comment:
maxTasks cant be provided as it is since there are some already created which might get over the creation of maxTasks. Anyways the idea behind maxTasks to provide an approximate of the max number of tasks to be created. Since log Tasks cannot be created in subsequent run for current run's finished tasks, it is best not to worry using the traskTracker for creating log tasks anywhere, and just add them to the table level or function level task tracker or loadTableTaskTracker, if that is present